### PR TITLE
Pattern baudrate take two

### PR DESCRIPTION
**Pull request by hs-viktor**
This should subsume and polish #145 (which I will close along with #74)

### CODE REVIEW

File: System/Posix/Terminal/ByteString.hsc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE PatternSynonyms #-}
```
Comment (hs-viktor) [inline System/Posix/Terminal/ByteString.hsc:2]: Actually, this pragma is not needed here, the synonyms are bundled in the defining module...  I'll revert this after CI passes, or if a fixup is needed to get it to pass.

Comment (hs-viktor) [reply System/Posix/Terminal/ByteString.hsc]: Done.

**Comment by hs-viktor** [comment]
Removed `#ifdefs` from the POSIX-mandated speeds, and extended/added haddock comments.

**Review by Bodigrim — APPROVED** [review]

**Comment by Bodigrim** [comment]
Looks great, please go ahead with a merge.

**Pull request merged**